package.json

```diff
@@ -71,24 +72,25 @@
 				"icon": {
 					"light": "./media/Preview.svg",
 					"dark": "./media/Preview_inverse.svg"
-				}
+				},
+				"when": "!virtualWorkspace && shellExecutionSupported"
```

Comment:
Not sure if this is working... ideally, this command should not be available in a browser environment (because the current implementation relies on a binary `wkhtmltopdf`)

Reply:
I thought we could just use `!isWeb`? Alas, it did not work.

The docs said:


I also tried `!virtualWorkspace` (noting [this](https://github.com/microsoft/vscode/issues/121165) might provide a clue why it is not working), or perhaps [this](https://github.com/microsoft/vscode/issues/124594) which might suggest it can't be used for debugging.

But presumably on the desktop client we could still have a virtualWorkspace? See https://code.visualstudio.com/api/extension-guides/virtual-workspaces.

I thought !isWeb 

I can't see a solution at this time.

Reply:
Finally, I tried:

`"workspacePlatform != webworker"` which also did not work... :pouting_cat: 

When I use the [Inspect Context Keys](
https://code.visualstudio.com/api/references/when-clause-contexts#inspect-context-keys-utility) utility, I can find some interesting variables:

In the web browser running on Linux.

```js
isLinux: true
isMac: false
isMacNative: false
isWeb: true
isWindows: false
virtualWorkspace: ""
workspacePlatform: "webworker"
```

`shellExecutionSupported` doesn't appear to exist at all.

Comparing with local installation running in vs code on Linux:

```js
isLinux: true
isMac: false
isMacNative: false
isWeb: false
isWindows: false
shellExecutionSupported: true
virtualWorkspace: ""
workspacePlatform: "linux"
```

I think the debug environment for the web extension perhaps doesn't support this for some reason. 

I think we should use:

```js
"when": "!web",
```
on the following commands initially to support a MVP:

```js
"command": "asciidoc.exportAsPDF",
"command": "asciidoc.pasteImage",
"command": "asciidoc.saveHTML",
"command": "asciidoc.saveDocbook",
```

Perhaps we'll have to do a trial release to verify this.

Reply:
Done!